From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: msizanoen <msizanoen@qtmlabs.xyz>
Date: Fri, 31 May 2024 04:58:24 +0200
Subject: Disable custom Vulkan loader on Linux


diff --git a/src/common/vulkan/BUILD.gn b/src/common/vulkan/BUILD.gn
index 87d2f5a682b141bf57804136fc7a701dcf5f558b..69067795171ab2db83b9ba17140a7a96b8831db7 100644
--- a/src/common/vulkan/BUILD.gn
+++ b/src/common/vulkan/BUILD.gn
@@ -10,7 +10,7 @@ if (angle_use_wayland) {
 }
 
 declare_args() {
-  angle_use_custom_libvulkan = !is_android && !is_fuchsia
+  angle_use_custom_libvulkan = !is_android && !is_fuchsia && !is_linux
 
   # On real CrOS devices, we should use the system Vulkan loader. On the
   # ChromeOS-for-Linux build, we should use the built libvulkan because the
diff --git a/src/common/vulkan/libvulkan_loader.cpp b/src/common/vulkan/libvulkan_loader.cpp
index f1b7e30d5b1efcf77cf1a8ceaf53f930cae8f93d..46e55516046e2fb04bb5b8a8311dc37af626601c 100644
--- a/src/common/vulkan/libvulkan_loader.cpp
+++ b/src/common/vulkan/libvulkan_loader.cpp
@@ -31,7 +31,7 @@ void *OpenLibVulkan()
     };
 
     constexpr SearchType kSearchTypes[] = {
-// On Android and Fuchsia we use the system libvulkan.
+// On Linux, Android and Fuchsia we use the system libvulkan.
 #if defined(ANGLE_USE_CUSTOM_LIBVULKAN)
         SearchType::ModuleDir,
 #else
