From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: msizanoen <msizanoen@qtmlabs.xyz>
Date: Sun, 24 Mar 2024 10:12:40 +0100
Subject: Revert "egl: Add logic to select preferred display"

This reverts commit ec6d62886345a11310bc01852c88ac0428c129cb.

diff --git a/DEPS b/DEPS
index 2803818988a8e99d2ed8361820437878b008a92f..451b65789d64243d35d657d239ac7342dd278617 100644
--- a/DEPS
+++ b/DEPS
@@ -723,7 +723,7 @@ deps = {
   # We use the upstream/main branch.
   'third_party/libdrm': {
     'url': Var('chromium_git') + '/chromiumos/third_party/libdrm@474894ed17a037a464e5bd845a0765a50f647898',
-    'condition': 'checkout_angle_mesa or not build_with_chromium',
+    'condition': 'checkout_angle_mesa',
   },
 
   # libjpeg_turbo is used by glmark2.
diff --git a/src/libANGLE/renderer/gl/BUILD.gn b/src/libANGLE/renderer/gl/BUILD.gn
index d421339146af1c586a9a92e04b7c2f193e6fc91a..7fa3d8636ab92fab77e9c886a03c8a915dae6dbd 100644
--- a/src/libANGLE/renderer/gl/BUILD.gn
+++ b/src/libANGLE/renderer/gl/BUILD.gn
@@ -38,9 +38,6 @@ config("angle_gl_backend_config") {
       ]
     }
   }
-  if (angle_has_build && (is_linux || is_chromeos)) {
-    defines += [ "ANGLE_HAS_LIBDRM" ]
-  }
   include_dirs = [ "../../../third_party/khronos" ]
 }
 
@@ -55,10 +52,6 @@ angle_source_set("angle_gl_backend") {
     "$angle_root:angle_image_util",
   ]
 
-  if (angle_has_build && (is_linux || is_chromeos)) {
-    deps += [ "//build/config/linux/libdrm" ]
-  }
-
   if (is_win) {
     deps += [ "$angle_root:angle_d3d_format_tables" ]
   }
diff --git a/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp b/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
index 39b6f035093379a9c967ea4c77b6707f2ef73620..7b97692c0b0e68ff01dcbddbc6e8943442bd900f 100644
--- a/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
+++ b/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
@@ -16,13 +16,6 @@
 #include "libANGLE/renderer/gl/FunctionsGL.h"
 #include "libANGLE/renderer/gl/egl/functionsegl_typedefs.h"
 
-#if defined(ANGLE_HAS_LIBDRM)
-// clang-format off
-#include <fcntl.h>
-#include <xf86drm.h>
-// clang-format on
-#endif  // defined(ANGLE_HAS_LIBDRM)
-
 namespace
 {
 
@@ -238,26 +231,13 @@ egl::Error FunctionsEGL::initialize(EGLAttrib platformType, EGLNativeDisplayType
     ANGLE_GET_PROC_OR_ERROR(&mFnPtrs->surfaceAttribPtr, eglSurfaceAttrib);
     ANGLE_GET_PROC_OR_ERROR(&mFnPtrs->swapIntervalPtr, eglSwapInterval);
 
-    // Querying EGL_EXTENSIONS string and loading it into the mExtensions
-    // vector will at this point retrieve the client extensions since mEGLDisplay is still
-    // EGL_NO_DISPLAY. This is desired, and mExtensions will later be reinitialized with the display
-    // extensions once the display is created and initialized.
-    queryExtensions();
-
-#if defined(ANGLE_HAS_LIBDRM)
-    mEGLDisplay = getPreferredDisplay(&majorVersion, &minorVersion);
-#endif  // defined(ANGLE_HAS_LIBDRM)
-
-    if (mEGLDisplay == EGL_NO_DISPLAY)
+    if (IsValidPlatformTypeForPlatformDisplayConnection(platformType))
     {
-        if (IsValidPlatformTypeForPlatformDisplayConnection(platformType))
-        {
-            mEGLDisplay = getPlatformDisplay(platformType, nativeDisplay);
-        }
-        else
-        {
-            mEGLDisplay = mFnPtrs->getDisplayPtr(nativeDisplay);
-        }
+        mEGLDisplay = getPlatformDisplay(platformType, nativeDisplay);
+    }
+    else
+    {
+        mEGLDisplay = mFnPtrs->getDisplayPtr(nativeDisplay);
     }
 
     if (mEGLDisplay != EGL_NO_DISPLAY &&
@@ -289,10 +269,12 @@ egl::Error FunctionsEGL::initialize(EGLAttrib platformType, EGLNativeDisplayType
 
     ANGLE_GET_PROC_OR_ERROR(&mFnPtrs->getCurrentContextPtr, eglGetCurrentContext);
 
-    if (!queryExtensions())
+    const char *extensions = queryString(EGL_EXTENSIONS);
+    if (!extensions)
     {
         return egl::Error(mFnPtrs->getErrorPtr(), "Failed to query extensions in system egl");
     }
+    angle::SplitStringAlongWhitespace(extensions, &mExtensions);
 
     if (hasExtension("EGL_KHR_image_base"))
     {
@@ -389,22 +371,20 @@ egl::Error FunctionsEGL::terminate()
     return egl::Error(mFnPtrs->getErrorPtr());
 }
 
-bool FunctionsEGL::queryExtensions()
-{
-    const char *extensions = queryString(EGL_EXTENSIONS);
-    if (!extensions)
-    {
-        return false;
-    }
-
-    angle::SplitStringAlongWhitespace(extensions, &mExtensions);
-
-    return true;
-}
-
 EGLDisplay FunctionsEGL::getPlatformDisplay(EGLAttrib platformType,
                                             EGLNativeDisplayType nativeDisplay)
 {
+    // As in getNativeDisplay(), querying EGL_EXTENSIONS string and loading it into the mExtensions
+    // vector will at this point retrieve the client extensions since mEGLDisplay is still
+    // EGL_NO_DISPLAY. This is desired, and mExtensions will later be reinitialized with the display
+    // extensions once the display is created and initialized.
+    const char *extensions = queryString(EGL_EXTENSIONS);
+    if (!extensions)
+    {
+        return EGL_NO_DISPLAY;
+    }
+    angle::SplitStringAlongWhitespace(extensions, &mExtensions);
+
     PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplayEXTPtr;
     if (!hasExtension("EGL_EXT_platform_base") ||
         !SetPtr(&getPlatformDisplayEXTPtr, getProcAddress("eglGetPlatformDisplayEXT")))
@@ -428,35 +408,24 @@ EGLDisplay FunctionsEGL::getPlatformDisplay(EGLAttrib platformType,
                                     reinterpret_cast<void *>(nativeDisplay), nullptr);
 }
 
-std::vector<EGLDeviceEXT> FunctionsEGL::queryDevices(int *major, int *minor)
-{
-    // Only called after confirming we have the necessary extension.
-    PFNEGLQUERYDEVICESEXTPROC queryDevicesEXTPtr;
-    if (!SetPtr(&queryDevicesEXTPtr, getProcAddress("eglQueryDevicesEXT")))
-    {
-        return {};
-    }
-
-    // Get a list of native device objects.
-    const EGLint kMaxDevices = 32;
-    EGLint numDevices        = 0;
-    std::vector<EGLDeviceEXT> devices(kMaxDevices, EGL_NO_DEVICE_EXT);
-    if (!queryDevicesEXTPtr(kMaxDevices, devices.data(), &numDevices))
-    {
-        return {};
-    }
-
-    devices.resize(numDevices);
-    return devices;
-}
-
 EGLDisplay FunctionsEGL::getNativeDisplay(int *major, int *minor)
 {
+    // We haven't queried extensions yet since some platforms require a display
+    // to do so. We'll query them now since we need some for this fallback, and
+    // then re-query them again later once we have the display.
+    const char *extensions = queryString(EGL_EXTENSIONS);
+    if (!extensions)
+    {
+        return EGL_NO_DISPLAY;
+    }
+    angle::SplitStringAlongWhitespace(extensions, &mExtensions);
+
     // This fallback mechanism makes use of:
     // - EGL_EXT_device_enumeration or EGL_EXT_device_base for eglQueryDevicesEXT
     // - EGL_EXT_platform_base for eglGetPlatformDisplayEXT
     // - EGL_EXT_platform_device for EGL_PLATFORM_DEVICE_EXT
-    PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplayEXTPtr;
+    PFNEGLQUERYDEVICESEXTPROC queryDevices;
+    PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplay;
     bool hasQueryDevicesEXT =
         hasExtension("EGL_EXT_device_enumeration") || hasExtension("EGL_EXT_device_base");
     bool hasPlatformBaseEXT   = hasExtension("EGL_EXT_platform_base");
@@ -465,24 +434,30 @@ EGLDisplay FunctionsEGL::getNativeDisplay(int *major, int *minor)
     {
         return EGL_NO_DISPLAY;
     }
-
-    if (!SetPtr(&getPlatformDisplayEXTPtr, getProcAddress("eglGetPlatformDisplayEXT")))
+    if (!SetPtr(&queryDevices, getProcAddress("eglQueryDevicesEXT")) ||
+        !SetPtr(&getPlatformDisplay, getProcAddress("eglGetPlatformDisplayEXT")))
     {
         return EGL_NO_DISPLAY;
     }
 
-    std::vector<EGLDeviceEXT> devices = queryDevices(major, minor);
-    if (devices.size() == 0)
+    // Get a list of native device objects.
+    const EGLint kMaxDevices = 32;
+    EGLDeviceEXT eglDevices[kMaxDevices];
+    EGLint numDevices = 0;
+    if (!queryDevices(kMaxDevices, eglDevices, &numDevices))
     {
         return EGL_NO_DISPLAY;
     }
 
     // Look for the first native device that gives us a valid display.
-    for (const EGLDeviceEXT device : devices)
+    for (EGLint i = 0; i < numDevices; i++)
     {
-        EGLDisplay display = getPlatformDisplayEXTPtr(EGL_PLATFORM_DEVICE_EXT, device, nullptr);
-        if (mFnPtrs->getErrorPtr() == EGL_SUCCESS &&
-            mFnPtrs->initializePtr(display, major, minor) == EGL_TRUE)
+        EGLDisplay display = getPlatformDisplay(EGL_PLATFORM_DEVICE_EXT, eglDevices[i], nullptr);
+        if (mFnPtrs->getErrorPtr() != EGL_SUCCESS)
+        {
+            continue;
+        }
+        if (mFnPtrs->initializePtr(display, major, minor) == EGL_TRUE)
         {
             return display;
         }
@@ -491,103 +466,6 @@ EGLDisplay FunctionsEGL::getNativeDisplay(int *major, int *minor)
     return EGL_NO_DISPLAY;
 }
 
-#if defined(ANGLE_HAS_LIBDRM)
-EGLDeviceEXT FunctionsEGL::getPreferredEGLDevice(const std::vector<EGLDeviceEXT> &devices)
-{
-    // Only called after confirming we have the necessary extension.
-    PFNEGLQUERYDEVICESTRINGEXTPROC queryDeviceStringEXTPtr;
-    if (!SetPtr(&queryDeviceStringEXTPtr, getProcAddress("eglQueryDeviceStringEXT")))
-    {
-        return EGL_NO_DEVICE_EXT;
-    }
-
-    std::map<EGLDeviceEXT, std::string> device_drivers;
-    for (const EGLDeviceEXT device : devices)
-    {
-        const char *filename = queryDeviceStringEXTPtr(device, EGL_DRM_DEVICE_FILE_EXT);
-        if (!filename)
-        {
-            continue;
-        }
-
-        int fd = open(filename, O_RDWR);
-        if (!fd)
-        {
-            continue;
-        }
-
-        drmVersionPtr version = drmGetVersion(fd);
-        if (version)
-        {
-            const std::string driver_name(version->name, version->name_len);
-            device_drivers.insert({device, driver_name});
-        }
-
-        drmFreeVersion(version);
-        close(fd);
-    }
-
-    const char *preferred_drivers[3] = {"i915", "amdgpu", "virtio_gpu"};
-    for (const char *preferred_driver : preferred_drivers)
-    {
-        for (const EGLDeviceEXT device : devices)
-        {
-            const auto driver = device_drivers.find(device);
-            if (driver != device_drivers.end() && driver->second == preferred_driver)
-            {
-                return device;
-            }
-        }
-    }
-
-    return EGL_NO_DEVICE_EXT;
-}
-
-EGLDisplay FunctionsEGL::getPreferredDisplay(int *major, int *minor)
-{
-    // This  mechanism makes use of:
-    // - EGL_EXT_device_enumeration or EGL_EXT_device_base for eglQueryDevicesEXT
-    // - EGL_EXT_platform_base for eglGetPlatformDisplayEXT
-    // - EGL_EXT_platform_device for EGL_PLATFORM_DEVICE_EXT
-    // - EGL_EXT_device_query for eglQueryDeviceStringEXT
-    PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplayEXTPtr;
-    bool hasQueryDevicesEXT =
-        hasExtension("EGL_EXT_device_enumeration") || hasExtension("EGL_EXT_device_base");
-    bool hasPlatformBaseEXT   = hasExtension("EGL_EXT_platform_base");
-    bool hasPlatformDeviceEXT = hasExtension("EGL_EXT_platform_device");
-    bool hasDeviceQueryEXT    = hasExtension("EGL_EXT_device_query");
-    if (!hasQueryDevicesEXT || !hasPlatformBaseEXT || !hasPlatformDeviceEXT || !hasDeviceQueryEXT)
-    {
-        return EGL_NO_DISPLAY;
-    }
-
-    if (!SetPtr(&getPlatformDisplayEXTPtr, getProcAddress("eglGetPlatformDisplayEXT")))
-    {
-        return EGL_NO_DISPLAY;
-    }
-
-    std::vector<EGLDeviceEXT> devices = queryDevices(major, minor);
-    if (devices.size() == 0)
-    {
-        return EGL_NO_DISPLAY;
-    }
-
-    EGLDeviceEXT device = getPreferredEGLDevice(devices);
-    if (device == EGL_NO_DEVICE_EXT)
-    {
-        return EGL_NO_DISPLAY;
-    }
-
-    EGLDisplay display = getPlatformDisplayEXTPtr(EGL_PLATFORM_DEVICE_EXT, device, nullptr);
-    if (mFnPtrs->getErrorPtr() == EGL_SUCCESS)
-    {
-        return display;
-    }
-
-    return EGL_NO_DISPLAY;
-}
-#endif  // defined(ANGLE_HAS_LIBDRM)
-
 class FunctionsGLEGL : public FunctionsGL
 {
   public:
diff --git a/src/libANGLE/renderer/gl/egl/FunctionsEGL.h b/src/libANGLE/renderer/gl/egl/FunctionsEGL.h
index c3f5bd065da8db703e9795724f7b95c10a4f4057..b93ea49e6b79981b864a186f42f5c6007962c161 100644
--- a/src/libANGLE/renderer/gl/egl/FunctionsEGL.h
+++ b/src/libANGLE/renderer/gl/egl/FunctionsEGL.h
@@ -129,23 +129,9 @@ class FunctionsEGL
     // Helper mechanism for creating a display for the desired platform type.
     EGLDisplay getPlatformDisplay(EGLAttrib platformType, EGLNativeDisplayType nativeDisplay);
 
-    // Helper method to populate EGL extensions. Returns true if extensions are found.
-    bool queryExtensions();
-
-    // Helper method to query egl for devices.
-    std::vector<EGLDeviceEXT> queryDevices(int *major, int *minor);
-
     // Fallback mechanism for creating a display from a native device object.
     EGLDisplay getNativeDisplay(int *major, int *minor);
 
-#if defined(ANGLE_HAS_LIBDRM)
-    // Helper method to select preferred EGL device from queried devices.
-    EGLDeviceEXT getPreferredEGLDevice(const std::vector<EGLDeviceEXT> &devices);
-
-    // Helper method to get preferred display.
-    EGLDisplay getPreferredDisplay(int *major, int *minor);
-#endif  // defined(ANGLE_HAS_LIBDRM)
-
     struct EGLDispatchTable;
     EGLDispatchTable *mFnPtrs;
     EGLDisplay mEGLDisplay;
