From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Myrrh Periwinkle <myrrhperiwinkle@qtmlabs.xyz>
Date: Tue, 17 Jun 2025 10:42:46 +0700
Subject: Revert "Add a missing #include."

This reverts commit 2ad7b23b13aef70384a3ffe641ea28bc12f83dec.

diff --git a/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp b/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
index 58e8897387604a9b7b8ed727f140f9a89e8b91ff..0f2567ab7ab1d3012297a6daacb41e92a1b6c250 100644
--- a/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
+++ b/src/libANGLE/renderer/gl/egl/FunctionsEGL.cpp
@@ -19,7 +19,6 @@
 #if defined(ANGLE_HAS_LIBDRM)
 // clang-format off
 #include <fcntl.h>
-#include <unistd.h>
 #include <xf86drm.h>
 // clang-format on
 #endif  // defined(ANGLE_HAS_LIBDRM)
