From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: msizanoen <msizanoen@qtmlabs.xyz>
Date: Thu, 6 Jul 2023 13:14:00 +0200
Subject: Vulkan: misc fixes for validation errors


diff --git a/src/libANGLE/renderer/vulkan/vk_helpers.cpp b/src/libANGLE/renderer/vulkan/vk_helpers.cpp
index 702d503ca8ae204bf960b0040c6ff8930f5e9f0f..edb75967e86a744c46c6a67d7723a0725673a11a 100644
--- a/src/libANGLE/renderer/vulkan/vk_helpers.cpp
+++ b/src/libANGLE/renderer/vulkan/vk_helpers.cpp
@@ -5965,6 +5965,9 @@ const void *ImageHelper::DeriveCreateInfoPNext(
     (*imageListFormatsStorage)[0] = vk::GetVkFormatFromFormatID(actualFormatID);
     (*imageListFormatsStorage)[1] = vk::GetVkFormatFromFormatID(additionalFormat);
 
+    *imageFormatListInfoStorage       = {};
+    imageFormatListInfoStorage->sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_LIST_CREATE_INFO_KHR;
+
     if (renderer->getFeatures().supportsImageFormatList.enabled &&
         renderer->haveSameFormatFeatureBits(actualFormatID, additionalFormat))
     {
@@ -5973,8 +5976,7 @@ const void *ImageHelper::DeriveCreateInfoPNext(
 
         // There is just 1 additional format we might use to create a VkImageView for this
         // VkImage
-        imageFormatListInfoStorage->sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_LIST_CREATE_INFO_KHR;
-        imageFormatListInfoStorage->pNext = pNext;
+        imageFormatListInfoStorage->pNext           = pNext;
         imageFormatListInfoStorage->viewFormatCount = kImageListFormatCount;
         imageFormatListInfoStorage->pViewFormats    = imageListFormatsStorage->data();
 
